Moneypool

JohannaSchmidt opened the pull request.

Commit:
first try of moneypool rest adapter

- added controller and service for creating and
retrieving moneypools

Change-Id: I72e9da451574b8f7d3348e0ff63e6584218115d8

Commit:
Merge remote-tracking branch 'origin/master' into moneypool

Change-Id: Ib3a7f1a4067bb99249c389b2ef8d8a8221471cd8

Commit:
Moneypool

tried to connect the rest adapter with
database
--> post endpoint for creating a new moneypool
--> get request for retrieving a moneypool

Change-Id: Ifbd1381c92fa7f30f80e1eaf68f8a9373c3d7b7f
Problem: retrieving the right user information

d4l-w4r commented on src/main/kotlin/com/wirvsvirus/solipool/moneypool/MoneypoolService.kt:
```diff
@@ -0,0 +1,25 @@
+        private val accountBalanceService: AccountBalanceService,
+        private val moneypoolRepository: MoneypoolRepository) {
+
+    fun getMoneypool(id: UUID): Moneypool {
```
Würde ich nullable machen

d4l-w4r commented on src/main/kotlin/com/wirvsvirus/solipool/moneypool/MoneypoolService.kt:
```diff
@@ -0,0 +1,25 @@
+    fun getMoneypool(id: UUID): Moneypool {
+        val moneypoolRow = moneypoolRepository.getMoneypool(id)
+        val balance = accountBalanceService.getBalanceForPool(id) ?: BigInteger.ZERO
+        return Moneypool(moneypoolRow?.id?.value ?: UUID.randomUUID(), moneypoolRow?.name ?: "", 1, balance)
```
return if (moneypoolRow != null && balance != null) {
    Moneypool(moneypoolRow.id.value, moneypoolRow.name, moneypoolRow.creator.id, balance)
}

d4l-w4r commented on src/main/kotlin/com/wirvsvirus/solipool/banking/AccountBalanceService.kt:
```diff
@@ -8,3 +9,10 @@ interface AccountBalanceService {
+
+@Service
+class AccountBalanceServiceImpl : AccountBalanceService {
+    override fun getBalanceForPool(id: UUID): BigInteger? {
```
Heads up: Das interface hab ich verkackt. Wir wollen natürlich `BigDecimal` benutzen, damit wir auch Cent Beträge darstellen können :D

d4l-w4r commented on src/main/kotlin/com/wirvsvirus/solipool/moneypool/model/Moneypool.kt:
```diff
@@ -0,0 +1,6 @@
+import java.math.BigInteger
+import java.util.*
+
+data class Moneypool(val id: UUID, val name: String, val creator: Int, val balance: BigInteger)
```
entsprechend dann auch hier BigDecimal

Commit:
Moneypool

tried to connect the rest adapter with
database
--> post endpoint for creating a new moneypool
--> get request for retrieving a moneypool

Change-Id: I0dbdb33c07293c83f83d09f3e2368788eb79b119
Problem: retrieving the right user information

Commit:
Moneypool

Wrapped creator request in transaction to get
right user

Change-Id: I8eccb51b7bfbe114497cda2d88763c961886741b

d4l-w4r committed:
Merge branch 'master' into moneypool
